ext/session/session.c

```diff
@@ -2066,7 +2066,7 @@ static PHP_FUNCTION(session_set_save_handler)
 		}
 	}
 
-	if (PS(mod) && PS(mod) != &ps_mod_user) {
+	if (!PS(mod) || PS(mod) != &ps_mod_user) {
```

Comment:
Maybe extract this whole code block into a function, as it's used twice?

Reply:
Good idea, but naming things is hard. How about `set_user_save_handler_ini()`?

Reply:
Sounds fine to me!

Reply:
Great, so I went with this.